x-pack/plugins/cases/server/connectors/cases/cases_connector.ts

```diff
@@ -418,6 +458,115 @@ export class CasesConnector extends SubActionConnector<
       .join(' and ');
   }
 
+  private async handleClosedCases(
+    params: CasesConnectorRunParams,
+    casesClient: CasesClient,
+    casesMap: Map<string, GroupedAlertsWithCases>
+  ) {
+    const entriesWithClosedCases = Array.from(casesMap.values()).filter(
+      (theCase) => theCase.theCase.status === CaseStatuses.closed
+    );
+
+    if (entriesWithClosedCases.length === 0) {
+      return casesMap;
+    }
+
+    const res = params.reopenClosedCases
+      ? await this.reopenClosedCases(casesClient, entriesWithClosedCases, casesMap)
+      : await this.createNewCasesOutOfClosedCases(
+          params,
+          casesClient,
+          entriesWithClosedCases,
+          casesMap
+        );
+
+    /**
+     * The initial map contained the closed cases. We need to remove them to
+     * avoid attaching alerts to a close case
+     */
+    return new Map([...res].filter(([_, record]) => record.theCase.status !== CaseStatuses.closed));
+  }
+
+  private async reopenClosedCases(
+    casesClient: CasesClient,
+    closedCasesEntries: GroupedAlertsWithCases[],
+    casesMap: Map<string, GroupedAlertsWithCases>
+  ): Promise<Map<string, GroupedAlertsWithCases>> {
+    const casesMapWithClosedCasesOpened = new Map(casesMap);
+
+    const bulkUpdateReq = closedCasesEntries.map((entry) => ({
+      id: entry.theCase.id,
+      version: entry.theCase.version,
+      status: CaseStatuses.open,
+    }));
+
+    /**
+     * TODO: bulkUpdate throws an error. Retry on errors.
+     */
+    const bulkUpdateCasesResponse = await casesClient.cases.bulkUpdate({ cases: bulkUpdateReq });
+
+    for (const res of bulkUpdateCasesResponse) {
+      if (casesMap.has(res.id)) {
+        const data = casesMap.get(res.id) as GroupedAlertsWithCases;
+        casesMapWithClosedCasesOpened.set(res.id, { ...data, theCase: res });
+      }
+    }
+
+    return casesMapWithClosedCasesOpened;
+  }
+
+  private async createNewCasesOutOfClosedCases(
+    params: CasesConnectorRunParams,
+    casesClient: CasesClient,
+    closedCasesEntries: GroupedAlertsWithCases[],
+    casesMap: Map<string, GroupedAlertsWithCases>
+  ): Promise<Map<string, GroupedAlertsWithCases>> {
+    const casesMapWithClosedCasesOpened = new Map(casesMap);
+    const casesMapAsArray = Array.from(casesMap.values());
+
+    const findEntryByOracleRecord = (oracleId: string) => {
+      return casesMapAsArray.find((record) => record.oracleRecord.id === oracleId);
+    };
+
+    const bulkUpdateOracleValidRecords = await this.increaseOracleRecordCounter(
```

Comment:
To create new cases, first we need to increase the counter to be able to get the new case ID.